src/middlewares/createInsightsMiddleware.ts

```diff
@@ -132,7 +138,13 @@ export const createInsightsMiddleware: CreateInsightsMiddleware = (props) => {
           if (onEvent) {
             onEvent(event, _insightsClient);
           } else if (event.insightsMethod) {
-            if (hasUserToken()) {
+            // At this point, instantSearchInstance must be started and
+            // it means there is a configure widget (added above).
+            const hasUserToken = Boolean(
+              instantSearchInstance.renderState[instantSearchInstance.indexName]
+                .configure!.widgetParams.searchParameters.userToken
```

Comment:
if the setUserTokenToSearch never was called, isn't configure undefined?

Reply:
We add it in the beginning for `clickAnalytics`
https://github.com/algolia/instantsearch.js/blob/6fba77c7167ee039ff6d3685616e7dfe64c81e8e/src/middlewares/createInsightsMiddleware.ts#L104:L106